src/api-client.ts

```diff
@@ -111,46 +80,47 @@ export class TradingSimulatorClient {
    * @returns A promise that resolves to the API response
    */
   private async request<T>(method: string, path: string, body: any = null): Promise<T> {
-    // Don't modify the path - use it exactly as provided
-    // This ensures the path used for signatures matches the actual request path
     const url = `${this.baseUrl}${path}`;
     
-    // Handle body consistently - stringify once and only once
-    let bodyString = '{}';
-    if (body !== null) {
-      bodyString = typeof body === 'string' ? body : JSON.stringify(body);
-    }
+    // Convert body to JSON string if it exists
+    const bodyString = body ? JSON.stringify(body) : undefined;
     
-    // Generate headers with the properly formatted body string
-    const headers = this.generateHeaders(method, path, bodyString);
+    // Generate headers with Bearer token authentication
+    const headers = this.generateHeaders();
     
     const options: RequestInit = {
       method: method.toUpperCase(),
       headers,
-      body: body !== null ? bodyString : undefined
+      body: bodyString
     };
 
+    if (this.debug) {
+      console.log('[ApiClient] Request details:');
+      console.log('[ApiClient] Method:', method);
+      console.log('[ApiClient] URL:', url);
+      console.log('[ApiClient] Body:', body ? JSON.stringify(body, null, 2) : 'none');
+    }
+
     try {
       const response = await fetch(url, options);
-      let data: any;
       
+      let data: any;
       try {
         const text = await response.text();
-        try {
-          data = JSON.parse(text);
-        } catch (parseError) {
-          throw new Error(`Failed to parse response: ${parseError}`);
-        }
+        data = JSON.parse(text);
       } catch (parseError) {
-        throw new Error(`Failed to read response: ${parseError}`);
+        throw new Error(`Failed to parse response: ${parseError}`);
       }
       
       if (!response.ok) {
```

Comment:
shouldn't this check happen before we try to parse the body?

Reply:
most definitely: [b1042c4](https://github.com/recallnet/trading-simulator-mcp/pull/2/commits/b1042c48c143de142638c7632c7d579a47e2a870)